File: examples/src/main/java/com/yetu/beanutils/Simple.java
package com.yetu.beanutils;

public class Simple {
    private final String name;
    private final int value;

    public Simple(String value) {
        this.name = value;
        this.value = 0;
    }

    public Simple() {
        this("", 0);
    }

    public Simple(String name, int value) {
        this.name = name;
        this.value = value;
    }

    public int getValue() {
        return value;
    }

    public boolean isEmpty() { return getFull().isEmpty(); }

    public boolean hasFoo() { return getFull().toLowerCase().contains("foo"); }

    public int getLength() { return getFull().length(); }

    public String getName() { return name; }

    public String getFull() { return name + ": " + value;}

    public String toString() { return getFull(); }
}

File: examples/src/main/scala/com/yetu/beanutils/companions/Simple.scala
package com.yetu.beanutils.companions

import com.yetu.beanutils.beanCompanion
import com.yetu.beanutils.{beans => b}
/**
 * Test object for the @beanCompanion macro
 */
@beanCompanion[b.Simple] object Simple {
  def foo = "Foo"
}
